packages/storybook/src/schematics/storybook-configure/storybook-configure.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  apply,
+  chain,
+  mergeWith,
+  move,
+  Rule,
+  schematic,
+  SchematicContext,
+  Tree,
+  url
+} from '@angular-devkit/schematics';
+import { getWorkspace } from '@schematics/angular/utility/config';
+import {} from '@schematics/angular/utility/json-utils';
+import { getProject } from '@schematics/angular/utility/project';
+import { updateWorkspace } from '@schematics/angular/utility/workspace';
+import { parseJsonAtPath } from '../../utils/utils';
+import { CypressConfigureSchema } from '../cypress-configure';
+import { StorybookStoriesSchema } from '../storybook-stories';
+import { StorybookConfigureSchema } from './schema';
+
+export default function(schema: StorybookConfigureSchema): Rule {
+  return chain([
+    createStorybookDir(schema.name),
+    configureTsConfig(schema.name),
+    addStorybookTask(schema.name),
+    schema.configureCypress
+      ? schematic<CypressConfigureSchema>('cypress-configure', {
+          name: schema.name
+        })
+      : () => {},
+    schema.generateStories
+      ? schematic<StorybookStoriesSchema>('storybook-stories', {
+          name: schema.name,
+          generateCypressSpecs:
+            schema.configureCypress && schema.generateCypressSpecs
+        })
+      : () => {}
+  ]);
+}
+
+function createStorybookDir(projectName: string): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    context.logger.debug('adding .storybook folder to lib');
+
+    return chain([
+      mergeWith(
+        apply(url('./files'), [move(getProject(tree, projectName).root)])
+      )
+    ])(tree, context);
+  };
+}
+
+function configureTsConfig(projectName: string): Rule {
+  return (tree: Tree) => {
+    const projectPath = getWorkspace(tree).projects[projectName].root;
+    const tsConfigPath = projectPath + '/tsconfig.lib.json';
+    const projectTsConfig = parseJsonAtPath(tree, tsConfigPath);
+
+    let tsConfigContent: any;
+
+    if (projectTsConfig && projectTsConfig.value) {
+      tsConfigContent = projectTsConfig.value;
+    } else {
+      return tree;
+    }
+
+    tsConfigContent.exclude = [...tsConfigContent.exclude, '**/*.stories.ts'];
+
+    return tree.overwrite(
+      tsConfigPath,
+      JSON.stringify(tsConfigContent, null, 2) + '\n'
+    );
+  };
+}
+
+function addStorybookTask(projectName: string): Rule {
+  return <any>updateWorkspace(workspace => {
+    workspace.projects.get(projectName).targets.add({
+      name: 'storybook',
```

Comment:
what happens if we want multiple projects to have storybook? Won't we have a conflict here with the name?

Reply:
Sorry, I should have included more information on the goal of this PR.  This schematic is set up to create an isolated storybook instance for a specific ui-lib.  For large monorepos, having a single storybook instance would be un-manageable.

This line doesn't cause a conflict - it creates a `storybook` target for the individual lib (just like each lib has its own `lint` target).  The `run-command` is set to point at the specific `.storybook` folder inside the lib (which only searches that lib for `*.stories.ts` files).

Reply:
Ok, that does explain a lot. But using storybook only for tests is a bit of waste I think. It would be great to have option of adding different addons, like np storysource etc from some cli prompt when crating story. 

I just integrated storybook in our company repo but went with one storybook per monorepo approach,  but I see some benefits from usage per lib.  

Have You tried how it works if u try to use affected? Does it run only required storybooks?

Reply:
nx affected doesn't work well for the storybook target, because it runs the first one and the process never finishes to run the next one.  It would make sense for a storybook-build target though.  I haven't written that, but it would be pretty straight forward to add.

nx affected works great for the cypress e2e tests though.  For each affected ui-lib, the storybook is launched, the cypress tests run, then all the processes end and the next e2e test target runs.

I would like to include more options for addons, that just hasn't made in this first pass.  You'll always be able to go into each `.storybook` folder and make whatever customizations you want.